scripts/build_with_vault.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+set -e
+
+MNT_PATH="couchbase"
+PLUGIN_NAME="vault-plugin-database-couchbase"
+PLUGIN_CATALOG_NAME="couchbase"
+
+#
+# Helper script for local development. Automatically builds and registers the
+# plugin. Requires `vault` is installed and available on $PATH.
+#
+
+# Get the right dir
+DIR="$(cd "$(dirname "$(readlink "$0")")" && pwd)"
+
+echo "==> Starting dev"
+
+echo "--> Scratch dir"
+echo "    Creating"
+SCRATCH="$DIR/tmp"
+mkdir -p "$SCRATCH/plugins"
+
+echo "--> Vault server"
+echo "    Writing config"
+tee "$SCRATCH/vault.hcl" > /dev/null <<EOF
+plugin_directory = "$SCRATCH/plugins"
+EOF
+
+echo "    Envvars"
+export VAULT_DEV_ROOT_TOKEN_ID="root"
+export VAULT_ADDR="http://127.0.0.1:8200"
+
```

Comment:
If you do add this it would be good to add `tmp/` to the gitignore and move

```
function cleanup {
  echo ""
  echo "==> Cleaning up"
  kill -INT "$VAULT_PID"
  rm -rf "$SCRATCH"
}
trap cleanup EXIT SIGINT
```
to the top of this script